app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -122,7 +125,8 @@ class WireActivityViewModel @Inject constructor(
     private val observeScreenshotCensoringConfigUseCaseProviderFactory: ObserveScreenshotCensoringConfigUseCaseProvider.Factory,
     private val globalDataStore: Lazy<GlobalDataStore>,
     private val observeIfE2EIRequiredDuringLoginUseCaseProviderFactory: ObserveIfE2EIRequiredDuringLoginUseCaseProvider.Factory,
-    private val workManager: Lazy<WorkManager>
+    private val workManager: Lazy<WorkManager>,
+    private val networkStateObserver: Lazy<NetworkStateObserver>
```

Comment:
this is over kill you don't need to check the network state, we can do the request to get the json with custom server meta data and check the error returned it is a network error with something list Io Exception then display this error

Reply:
That is unfortunately not doable:

- When we dont have internet connection and the config is valid we're getting:
`GenericError(cause=java.net.UnknownHostException: Unable to resolve host "nginz-https.bund-next-column-offline-android.wire.link": No address associated with hostname)`

- When we have internet connection and the config is invalid we're getting:
`GenericError(cause=java.net.UnknownHostException: Unable to resolve host "nginz-https.unreachable.wire.link": No address associated with hostname)`

So we're unable to know if the config was wrong or not with the exception we're getting

Reply:
Maybe we could slightly change the copy and use this dialog in both cases? 
Like: "You don't seem to be connected to the Internet or the config link you used is broken. Please check your Internet connection and try again." 😄

Reply:
Well yeah we could have only one for malformed json and no internet connection :D that'd solve the issue too

Reply:
> When we dont have internet connection and the config is valid we're getting:
GenericError(cause=java.net.UnknownHostException: Unable to resolve host "nginz-https.bund-next-column-offline-android.wire.link": No address associated with hostname)

I feel like i am missing something?
this will never happen how was the app able to know that the constant is valid if we are not able to read the content, since there is no internet?

Reply:
Changes pushed as we talked today, we now have the `no network` message for both cases ;)